net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/MetadataFactory.java

```diff
@@ -155,6 +220,176 @@ public static RuleMetadata getRuleMetadata(String name) {
+      final String[] packageTokens = moduleDetails.getFullQualifiedName().split("\\.");
+      group = getRuleGroupMetadata(sPackageToGroupNameMap
+              .get(packageTokens[packageTokens.length - 2]));
+    }
+    // third party extension modules
+    else {
+      String lookupKey = findLookupKey(moduleClassName);
+      String ruleGroupName = sThirdPartyRuleGroupMap.get(lookupKey).get("name");
+      group = getRuleGroupMetadata(ruleGroupName);
+      // if the group of the new check hasn't been formed yet
+      // and put into the sRuleGroupMetadata map
+      if (group == null) {
+        final String ruleDescription = sThirdPartyRuleGroupMap.get(lookupKey).get("description");
+        final int rulePriority = Integer.parseInt(sThirdPartyRuleGroupMap.get(lookupKey)
+                .get("priority"));
+        group = new RuleGroupMetadata(ruleGroupName, ruleDescription, false, rulePriority);
+        sRuleGroupMetadata.put(ruleGroupName, group);
+      }
+    }
+
+    RuleMetadata ruleMeta = new RuleMetadata(moduleDetails.getName(), moduleDetails.getName(),
+            moduleDetails.getParent(), MetadataFactory.getDefaultSeverity(),
+            false, true, true, false, group);
+    ruleMeta.setDescription(moduleDetails.getDescription());
+    moduleDetails.getProperties().forEach(modulePropertyDetails -> ruleMeta.getPropertyMetadata()
+            .add(createPropertyConfig(moduleDetails, modulePropertyDetails)));
+
+    return ruleMeta;
+  }
+
+  /**
+   * Generate all prefix strings from the packageName and find which is a valid key in
+   * {@code sThirdPartyRuleGroupMap}.
+   */
+  private static String findLookupKey(String packageName) {
+    final String[] packageTokens = packageName.split("\\.");
+    List<String> prefixList = new ArrayList<>();
+    String lookupKey = packageTokens[0];
+    prefixList.add(lookupKey);
+    for (int i = 1; i < packageTokens.length; i++) {
+      lookupKey += "." + packageTokens[i];
+      prefixList.add(lookupKey);
+    }
+
+    String result = null;
+    Collections.reverse(prefixList);
+    for (String candidate : prefixList) {
+      if (sThirdPartyRuleGroupMap.containsKey(candidate)) {
+        result = candidate;
+        break;
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Update the existing eclipse files based metadata with checkstyle files metadata, updating
+   * only those fields which are provided by checkstyle.
+   *
+   * @param ruleMetadata module rule metadata by eclipse
+   * @param moduleDetails module metadata by checkstyle
+   * @return updated rule metadata
+   */
+  public static RuleMetadata updateRuleMetadata(RuleMetadata ruleMetadata,
+          ModuleDetails moduleDetails) {
+    RuleMetadata modifiedRuleMetadata = new RuleMetadata(
+            ruleMetadata.getRuleName(), moduleDetails.getFullQualifiedName(),
+            moduleDetails.getParent(), ruleMetadata.getDefaultSeverityLevel(),
```

Comment:
`moduleDetails.getParent()` will return the full qualified name of the parent, while `RuleMetadata` expects the simple name of the parent. This breaks the plugin.